core/main.py

```diff
@@ -142,12 +141,22 @@ def dispatch(input, kind, func, args, autohelp=False):
         input.reply(func.__doc__)
         return
 
-    if hasattr(func, '_apikey'):
-        key = bot.config.get('api_keys', {}).get(func._apikey, None)
-        if key is None:
-            input.reply('error: missing api key')
-            return
-        input.api_key = key
+    if hasattr(func, '_apikeys'):
+        bot_keys = bot.config.get('api_keys', {})
+        if len(func._apikeys) > 1:
+            keys = {key: bot_keys.get(key) for key in func._apikeys}
+
+            missing = [keyname for keyname, value in keys.items() if value is None]
+            if missing:
+                input.reply('error: missing api keys - {}'.format(missing))
+                return
+        else:
```

Comment:
Instead of doing the if and having two separate retrieval branches

```
# fetch keys here
# check for missing keys and error

if len(keys) == 1:
  input.api_key = keys[0]
else:
  input.api_key = keys
```

Do you think that'd make it easier to read / follow?

Reply:
We'd have to do `keys.values()[0]`, but that would work and would be simpler.